bin/aurcheck

```diff
@@ -143,15 +158,33 @@ case $mode in
             tee >"$tmp"
         fi
 
-        cut -f1 "$tmp" | namever_aur | sort_join - "$tmp" | \
+        declare -a aur=() vcs=()
+        while read -r line; do
+            pkg=$(cut -f1 <<< "$line")
+            case "${pkg##*-}" in
+                bzr|git|hg|svn)
+                    vcs+=("$pkg")
+                    echo "$line" >> "$tmp_vcs"
+                    ;;
+                *)
+                    aur+=("$pkg")
+                    echo "$line" >> "$tmp_aur"
+                    ;;
+            esac
+        done < "$tmp"
+
+        echo "${aur[@]}" | namever_aur | sort_join - "$tmp_aur" | \
+            sort_by "$pkg_key" - | cmp_all
+
+        namever_vcs "${vcs[@]}" | sort_join - "$tmp_vcs" | \
```

Comment:
To describe the approach: split `$tmp` into AUR and VCS arrays, handle AUR ones just like before, but use the `namever_vcs` magic on VCS packages to get the new versions, then sort, compare and print exactly as for AUR ones.